### DISCUSSION TITLE
Does symfony support html streaming?

### DISCUSSION THREAD

**Discussion by Haberkamp** [symfony/symfony#57862] [Q&A]
Hello, I wanted to ask if Symfony does support of html responses? I want to achieve something similar to what Next.js and Astro do.  Let's say I have an e-commerce product listing page. Symfony should first render a static shell very with placeholder products very quick and stream in the actual product data later. This should increase the perceived performance for users as they already get a response while Symfony is fetching product data from the database.

**Comment by MatTheCat** [comment]
Hello,

what you want isn’t streaming HTML but lazy-loading; take a look at `hinclude.js` integration: https://symfony.com/doc/7.2/templates.html#templates-hinclude
